### PR TITLE
add makefile steps to build a universal binary of the library

### PR DESCRIPTION
**Pull request by jamespanic**
Should fix https://github.com/uspgamedev/luasteam/issues/39

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -23,7 +23,9 @@ all:
 STEAM_LIB=sdk/redistributable_bin
 
 osx:
-	$(CXX) $(SRC) $(CPP_FLAGS) ${STEAM_LIB}/osx/libsteam_api.dylib ${THIRD_PARTY}/lib/libluajit-5.1.a -o $(OSX_OUT) -shared -fPIC $(OSX_FLAGS)
+	$(CXX) $(SRC) $(CPP_FLAGS) -arch arm64 ${STEAM_LIB}/osx/libsteam_api.dylib ${THIRD_PARTY}/lib/libluajit-5.1.a -o $(OSX_OUT).arm64 -shared -fPIC $(OSX_FLAGS)
+	$(CXX) $(SRC) $(CPP_FLAGS) -arch x86_64 ${STEAM_LIB}/osx/libsteam_api.dylib ${THIRD_PARTY}/lib/libluajit-5.1.a -o $(OSX_OUT).x86_64 -shared -fPIC $(OSX_FLAGS)
```
Comment (yancouto) [inline Makefile:27]: Nice, it can all run in one machine, I thought it would need two.

**Review by yancouto — APPROVED** [review]
This looks good, but I can't test right now because I don't have a Mac and Travis is no longer free.

File: Makefile (hunk 1 above)
Comment (jamespanic) [reply Makefile:27]: That's correct intel and arm macs can cross-compile to both archs.

**Comment by jamespanic** [comment]
> This looks good, but I can't test right now because I don't have a Mac and Travis is no longer free.
> 
> I'm hoping you tested this in your machine.

Yes this has been tested and well be going into a shipping game.

**Comment by yancouto** [comment]
Thank you! I merged this for whoever wants to build, but I can't provide the auto-built universal libraries anymore because Travis is no longer free. (#46)

**Comment by yancouto** [comment]
Actually I managed to get free credits in Travis and fix the build (it was broken due to old XCode...)

If you have any free time, I would appreciate if you could quickly check if the CI-built universal library works: https://github.com/uspgamedev/luasteam/releases/tag/v3.2.1

**Comment by jamespanic** [comment]
```
> lipo -archs osx_luasteam.so
x86_64 arm64
```

looks good!

**Pull request merged**